yesod-form/Yesod/Form/Types.hs

```diff
@@ -80,6 +82,17 @@ instance Data.Traversable.Traversable FormResult where
       FormFailure errs -> pure (FormFailure errs)
       FormMissing -> pure FormMissing
 
+instance Alternative FormResult where
+    empty = FormMissing
+
+    (FormFailure e1) <|> (FormFailure _)  = FormFailure e1
```

Comment:
I think this can be more concisely expressed with just three clauses:

```haskell
FormFailure e <|> _ = FormFailure e
FormSuccess s <|> _ = FormSuccess s
FormMissing <|> f = f
```